src/analyze/analyze.c

```diff
@@ -2145,6 +2179,53 @@ static int do_condition(int argc, char *argv[], void *userdata) {
 }
 
 static int do_verify(int argc, char *argv[], void *userdata) {
+        char **filename;
+        _cleanup_free_ char *dst = NULL;
+        _cleanup_(rm_rf_and_freep) char *tempdir = NULL;
+        bool setup_alias = false;
+
+        STRV_FOREACH(filename, strv_skip(argv, 1))
+                if (strchr(*filename, ':')) {
+                        setup_alias = true;
+                        break;
+                }
+
+        if (setup_alias) {
+                int r, i = 1;
+
+                r = setup_working_dir(&tempdir);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to setup working directory");
+
+                STRV_FOREACH(filename, strv_skip(argv, 1)) {
+                        if (strchr(*filename, ':')) {
+                                _cleanup_free_ char *src = NULL, *argname = NULL;
+                                struct stat statbuf;
+
+                                argname = strdup(*filename);
+                                r = extract_first_word((const char **)filename, &src, ":", 0);
+                                if (r < 0 || !*filename)
+                                        return log_error_errno(r, "Failed to parse alias %s", argname);
+
+                                dst = path_join(tempdir, basename(*filename));
+                                if (!dst)
+                                        return log_oom();
+
+                                r = stat(src, &statbuf);
+                                if (r < 0)
+                                        return log_error_errno(errno, "Couldn't stat %s", src);
+
+                                r = copy_file(src, dst, 0, statbuf.st_mode, 0, 0, COPY_REFLINK);
+                                if (r < 0)
+                                        return log_error_errno(r, "Couldn't copy %s to alias %s", src, dst);
+
+                                argv[i] = dst;
```

Comment:
argv[i] is leaked, no? You'll need to build a strv and append `dst` in it at each round, so that these strings are all freed at the end (`_cleanup_strv_free_` to do that)

Reply:
This doesn't leak on my machine according to valgrind, but I can use a strv anyway, maybe it's implementation-specific